operator/pkg/compare/compare.go

```diff
@@ -253,6 +253,14 @@ func ManifestDiffWithRenameSelectIgnore(a, b, renameResources, selectResources,
 	}
 	bom := bo.ToMap()
 
+	if len(aom) == 0 && len(bom) == 0 {
```

Comment:
is an empty manifest really an error? with generic diff behavior, comparing two empty files just gives an empty diff without complaint.

Reply:
I did it this way because any valid YAML that isn't a Istio manifest has zero entries.  I didn't want to modify the code to pass around other statistics about it.

Agreed two empty text files are the same.

I make the claim that a file with 0 bytes or 0 resource definitions is NOT a manifest, and `istioctl manifest diff` should refuse to diff them.

Rather than following the semantics of Unix `diff` I am following the Javascript symantics:

```
> NaN == NaN
false
```

Reply:
@ostromart Was I able to convince you?

Reply:
sorry for the delay in the response - mentions don't seem to be reaching me atm. i'm not convinced that an empty istio manifest is invalid but i could well be wrong. @howardjohn wdyt? i'm ok with whichever way the tiebreaker goes.